optimum/intel/openvino/modeling_sam.py

```diff
@@ -0,0 +1,409 @@
+        else:
+            file_names = {}
+            for name, file_name in model_file_names.items():
+                model_cache_path = hf_hub_download(
+                    repo_id=model_id,
+                    filename=file_name,
+                    token=token,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    local_files_only=local_files_only,
+                )
+                file_names[name] = model_cache_path
+            model_save_dir = Path(model_cache_path).parent
+        if not compile_only:
+            vision_encoder_model = cls.load_model(file_names["vision_encoder_model"])
+            prompt_encoder_model = cls.load_model(file_names["prompt_encoder_mask_decoder_model"])
+        else:
+            vision_encoder_model = cls._compile_model(
+                file_names["vision_encoder_model"],
+                kwargs.get("device", "CPU"),
+                kwargs.get("ov_config"),
+                model_save_dir,
+            )
+            prompt_encoder_model = cls._compile_model(
+                file_names["prompt_encoder_mask_decoder_model"],
+                kwargs.get("device", "CPU"),
+                kwargs.get("ov_config"),
+                model_save_dir,
+            )
+
+        model = cls(
+            vision_encoder_model=vision_encoder_model,
+            prompt_encoder_mask_decoder_model=prompt_encoder_model,
+            config=config,
+            model_save_dir=model_save_dir,
+            **kwargs,
+        )
+
+        return model
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        task: Optional[str] = None,
+        use_cache: bool = True,
+        trust_remote_code: bool = False,
+        **kwargs,
+    ):
+        compile_only = kwargs.pop("compile_only", False)
+        if compile_only:
+            logger.warning(
+                "`compile_only` mode will be disabled because it does not support model export."
+                "Please provide openvino model obtained using optimum-cli or saved on disk using `save_pretrained`"
+            )
+            compile_only = False
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+
+        # This attribute is needed to keep one reference on the temporary directory, since garbage collecting
+        # would end-up removing the directory containing the underlying OpenVINO model
+        cls._model_save_dir_tempdirectory_instance = save_dir
```

Comment:
the tmpdir should be affected to the instance not the class, if I create two OVSamModels, the first will be lost and garbage collected

Reply:
moved